File: hash/leetcode_3/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        alphabet_set = set([])
        left = 0
        peak = 0

        for right in range(len(s)):
            while s[right] in alphabet_set:
                alphabet_set.remove(s[left])
                left += 1

            alphabet_set.add(s[right])
            
            peak = max(peak, right - left + 1)

        return peak

File: hash/leetcode_3/solution.js
/**
 * @param {string} s
 * @return {number}
 */

var lengthOfLongestSubstring = function(s) {
    const alphabetSet = new Set();
    let left = 0;
    let peak = 0;

    for (let right = 0; right < s.length; right++) {
        while (alphabetSet.has(s[right])) {
            alphabetSet.delete(s[left]);
            left++;
        }
        
        alphabetSet.add(s[right]);
        
        peak = Math.max(peak, right - left + 1);
    }

    return peak;
};
